quartz/components/pages/404.tsx

```diff
@@ -1,11 +1,14 @@
 import { i18n } from "../../i18n"
+import { pathToRoot } from "../../util/path"
 import { QuartzComponent, QuartzComponentConstructor, QuartzComponentProps } from "../types"
 
-const NotFound: QuartzComponent = ({ cfg }: QuartzComponentProps) => {
+const NotFound: QuartzComponent = ({ fileData, cfg }: QuartzComponentProps) => {
+  const baseDir = pathToRoot(fileData.slug!)
```

Comment:
`fileData.slug` doesn't work how you expect on 404 pages due to how 404 pages work :( 404 is served differently depending on provider (i.e. some replace the page with the content of the 404 page but keep the url the same, others redirect you to 404.html) so this link isn't guaranteed to be stable/resolve to the right place

normally we would use an absolute path for this but this requires the user to set baseUrl correctly

Reply:
Ah, I see :(.
Would using the `baseUrl` field in `quartz.config.ts` be acceptable, as this is an absolute URL?

Apologies, most of my previous tinkering with site generators has been through SvelteKit, which requires you to set the `basePath` at compile time.

Reply:
yeah should be ok to use baseUrl here imo!

Reply:
Have updated to use the BaseUrl config value. The link is based on the pathname portion after the domain name.
If the field is unset or doesn't contain a pathname, the link `href` is set to `/`.